### DISCUSSION TITLE
Localised production rules for several natural languages

### DISCUSSION THREAD

**Discussion by Felienne** [hedyorg/hedy#3113] [Language Ideas]
Since we are now supporting 30 very diverse languages (🤯!!), I think we should really start to think about localised productions. 

There are also some interesting research papers coming out demonstrating the problems of non-English kids in programming, for example: 

- https://www.tandfonline.com/doi/full/10.1080/08993408.2022.2079867 (Spanish)
- https://dl.acm.org/doi/abs/10.1145/3300115.3309521 (Chinese)

Let's use this discussion to keep track of what types of wishes and constraints some of the languages bring.

Here are some things I already know about:

- In Turkish, the keyword while should go before before a keyword (`i==0 iken`) rather than before it (`while i==0`)
- In standardised Chinese the keyword `in` should be split _around_ the list, so rather than <img width="236" alt="image" src="https://user-images.githubusercontent.com/1003685/181207287-4aad4b63-7bfc-4c5f-bfa9-fb4d345cfc96.png">
we want to have: <img width="270" alt="image" src="https://user-images.githubusercontent.com/1003685/181207323-8d5c0a9f-3ce9-416c-bb53-ca9b87d9e2d1.png">

Feel free to add any other things that your language or languages that you know might require, even if it feels really wild :)

**Comment by mjslee** [comment]
In Korean, there may be specific rules with number and command/keyword order. For example, in a modified" Level 1 - Turtle" example:

English:
`forward 50`
`turn left`
`turn right`

Korean:
`앞으로 50`
`왼쪽으로 돌아`
`오른쪽으로 돌아`

Literally in English:
`forward 50`
`(to)left turn`
`(to)right turn`

**Comment by mjslee** [comment]
I wonder if it might help to allow >1 keywords that do the same thing. For example (in English): we might have `print` and `output`.

In Korean, there may be a native Korean word and/or an equivalent sino-root (i.e., Chinese-root, called "hanja") word. Either word can typically be used, but there may be a more commonly used choice depending on the context.

Korean:
`앞으로 50`
`왼쪽으로 돌아`
`오른쪽으로 돌아`

Korean (with hanja-based words): 
`앞으로 50`
`좌로 회전`
`우로 회전`

Or written using hanja (which would **_not_** be common):
`앞으로 50`
`左로 迴轉`
`右로 迴轉`

Again, literally in English:
`forward 50`
`(to)left turn`
`(to)right turn`
